src/lib/core/ordered_latch.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2022, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#pragma once
+
+#include <assert.h>
+#include "fiber.h"
+#include "small/rlist.h"
+
+#if defined(__cplusplus)
+extern "C" {
+#endif /* defined(__cplusplus) */
+
+/**
+ * Latch of cooperative multitasking environment, which preserves strict order
+ * of fibers waiting for the latch.
+ */
```

Comment:
Why should one prefer to use `latch` instead of `ordered_latch` if they don't need ordering guarantees?

I thought that `ordered_latch` would have a different, less convenient API, but it looks like it has exactly the same API as `latch` and doesn't have any performance overhead in comparison to simple `latch`. Maybe, just rework the existing `latch` as this `ordered_latch`?

Reply:
Done